### PR TITLE
vsce: implement search UI

### PR DESCRIPTION
**Pull request by tjkandala**
Implement search UI in the VS Code extension.

Part of post-release cleanup tracked in #32674. 

NOTE: Only need review on changes _outside of_ `client/vscode`. We made [this decision](https://docs.google.com/document/d/1rC51DTAsiza2cnL1BYMEkgNxvISWnsDCmLOtnuI4vhM/edit#bookmark=id.4gncqez405s8) shortly after releasing v1 of the new VS Code extension. We do not have an automated release process yet, and the core codebase is mostly unaffected. The new GitHub Pull Request File Tree should help with this review :).

<details>
<summary>`search` + `search-ui` changes</summary>
<p>Mostly import updates</p>

</details>

<details>
<summary>`shared` changes</summary>
<p>Mostly extension host changes to enable running in VS Code</p>

</details>

<details>
<summary>`web` changes</summary>
<p>Mostly analytics and import updates</p>

</details>

## Test plan

- Web app should be unchanged functionally. Automated tests are passing. Manually tested search to ensure that the core flow still works.
  - Run `sg start web-standalone`
  - Execute a search
  - web search test

- VS Code extension has been released to users based on this branch multiple times. It is in a working state, confirmed through manual testing before each release. We have added a smoke test to ensure the extension compiles and loads (homepage), and we are in the process of implementing request interception in order to add integration tests.

## App preview:
- [Link](https://sg-web-tjk-vsce-search-ui.onrender.com)

### CODE REVIEW

File: client/web/src/tracking/eventLogger.ts [hunk 1]
```diff
@@ -273,6 +274,8 @@ function pageViewQueryParameters(url: string): EventQueryParameters {
         eventLogger.log('CodeMonitorEmailLinkClicked')
     } else if (utmSource === 'hubspot' && utmCampaign?.match(/^cloud-onboarding-email(.*)$/)) {
         eventLogger.log('UTMCampaignLinkClicked', utmProps, utmProps)
+    } else if (utmMedium === 'VSCIDE' && utmCampaign === 'vsce-sign-up') {
```
Comment (tjkandala) [inline client/web/src/tracking/eventLogger.ts:277]:
```suggestion
    } else if (utmMedium === 'VSCODE' && utmCampaign === 'vsce-sign-up') {
```

File: client/web/src/tracking/eventLogger.ts [hunk 2]
```diff
@@ -273,6 +274,8 @@ function pageViewQueryParameters(url: string): EventQueryParameters {
...
+        eventLogger.log('VSCIDESignUpLinkClicked', utmProps, utmProps)
```
Comment (tjkandala) [inline client/web/src/tracking/eventLogger.ts:278]:
```suggestion
        eventLogger.log('VSCODESignUpLinkClicked', utmProps, utmProps)
```

**Review by abeatrix — APPROVED** [review]
Reviewed the code for VSCE and LGTM!
Started a web instance and a VS Code instance and confirmed both are working accordingly.

File: client/web/src/SourcegraphWebApp.tsx [hunk 3]
```diff
@@ -564,4 +565,9 @@ export class SourcegraphWebApp extends React.Component<SourcegraphWebAppProps, S
             })
         )
     }
+    private fetchHighlightedFileLineRanges = (
```
Comment (limitedmage) [inline client/web/src/SourcegraphWebApp.tsx:568]: Can this be moved to a more relevant file and imported here? It seems way too specific for the root component to have this.

File: client/shared/src/components/FileMatch.tsx [hunk 4]
```diff
@@ -26,7 +26,7 @@ import { RepoIcon } from './RepoIcon'
 import { Props as ResultContainerProps, ResultContainer } from './ResultContainer'
 
 interface Props extends SettingsCascadeProps, TelemetryProps {
-    location: H.Location
+    location?: H.Location
```
Comment (limitedmage) [inline client/shared/src/components/FileMatch.tsx:29]: Is the `useLocation` hook an alternative here to remove the prop?

**Review by limitedmage — APPROVED** [review]
Reviewed search related code, looks good other than a couple of comments.

File: client/web/src/SourcegraphWebApp.tsx (hunk 3 above)
Comment (tjkandala) [reply client/web/src/SourcegraphWebApp.tsx:568]: I wish for the same thing, but I'm not sure how else we can pass the `fetchHighlightedFileLineRanges` with `platformContext` applied w/ a stable identity. 

Possibly a helper like `fetchHighlightedFileLineRangesWithContext`?

```ts
import { once } from 'lodash'

const fetchHighlightedFileLineRangesWithContext = once(
  (platformContext: PlatformContext) => 
    ({ parameters, force }) => 
      fetchHighlightedFileLineRanges({ ...parameters, platformContext }, force))
)

// In `SourcegraphWebApp.tsx`:
// ...
 fetchHighlightedFileLineRanges={fetchHighlightedFileLineRangesWithContext(this.platformContext)}
// ...
```

Comment (tjkandala) [reply client/web/src/SourcegraphWebApp.tsx:568]: I created https://github.com/sourcegraph/sourcegraph/issues/33856 to track this issue. There are some other methods on `SourcegraphWebApp` that could be moved out as part of this cleanup.

**Pull request merged**